File: LibSWBF2/FRAM.cpp
#include "stdafx.h"
#include "FRAM.h"

namespace LibSWBF2::Chunks::Mesh
{
	FRAM::FRAM()
	{

	}

	FRAM::~FRAM()
	{

	}

	void FRAM::RefreshSize()
	{
		Size = sizeof(int32_t) + sizeof(int32_t) + sizeof(float_t);
	}

	void FRAM::WriteToStream(FileWriter& stream)
	{
		BaseChunk::WriteToStream(stream);
		stream.WriteInt32(FrameRangeStart);
		stream.WriteInt32(FrameRangeEnd);
		stream.WriteFloat(FramesPerSecond);
	}

	void FRAM::ReadFromStream(FileReader& stream)
	{
		BaseChunk::ReadFromStream(stream);
		FrameRangeStart = stream.ReadInt32();
		FrameRangeEnd = stream.ReadInt32();
		FramesPerSecond = stream.ReadFloat();
	}
}

File: LibSWBF2.NET/Chunks/Mesh/FRAM.cs
using System.Runtime.InteropServices;

namespace LibSWBF2.Chunks.Mesh
{
    public unsafe class FRAM
    {
        [StructLayout(LayoutKind.Sequential)]
        internal struct NativeStruct
        {
            private BaseChunk.NativeStruct BaseChunk;
            public int FrameRangeStart;
            public int FrameRangeEnd;
            public float FramesPerSecond;
        }

        private NativeStruct* Ptr = null;

        public int FrameRangeStart
        {
            get { return Ptr->FrameRangeStart; }
            set { Ptr->FrameRangeStart = value; }
        }
        public int FrameRangeEnd
        {
            get { return Ptr->FrameRangeEnd; }
            set { Ptr->FrameRangeEnd = value; }
        }
        public float FramesPerSecond
        {
            get { return Ptr->FramesPerSecond; }
            set { Ptr->FramesPerSecond = value; }
        }

        internal FRAM(NativeStruct* ptr)
        {
            Ptr = ptr;
        }
    }
}
